Prepare Release v2.19.0

jwaspin opened the pull request:
Closes #142
Closes #222 
Closes #243 
Closes #240 
Closes #249 
Closes #255 
Closes #289 
Closes #260 
Closes #291 
Closes #292 
Closes #293 
Closes #294 
Closes #303 
Closes #304 
Closes #305 
Closes #314 
Closes #323 
Closes #320 

Might close the following:

#145 
#257 

### Testing

[Track testing deployment state here](https://docs.google.com/spreadsheets/d/1ghLMCOGO8nEzDMMRG80xkJujhJVSQi_RkqsQP7SAj4o)


https://dev-mdtranslator.mdeditor.org/

# Changes

Version 2.19.0 (beta.31)

- Fixed unit tests
- New GitHub action to run tests

## Dependencies

- mdJson-schemas updated to version 2.9.5
- mdCodes updated to version 2.10.0
- nokogiri updated to version 1.15

## Internal Object

- New newRangeElementDescription
- New newQualityResultFile
- Fixed newCoverageResult
- Fixed newDataQualityReport
- Fixed newStandaloneReport

## Readers

### fgdc

- Fixed the existing data quality reports implementation
- Renamed module_quality to module_dataQuality
- Disabling keywords requirement
- Refactored lineage / methodology / process handler

### mdJson

- Fixed version unit test
- Fixed & Completed Data Quality
  - Fixed & Completed Results (Conformance, Coverage, Descriptive, Qualitative)
  - Fixed reference to standaloneReport in the internal object (the reader and internal object were not aligned)
  - Fixed & Completed Quality Measure
  - New module_qualityResultFile
- Fixed order of legal / use constraints
- Fixed rangeElementDescription typo rangeElements -> rangeElement

### sbJson

- External Contact Identifiers
  - Restricted to "person" contact type
  - Looks for `orcId` and maps to `externalIdentifiers[ { identifier: <orcId value>, namespace: "ORCID", description: "Open Researcher and Contributor Id" } ]`

## Writers

### fgdc

- Renamed class_quality to class_dataQuality
- Fixed and Completed data quality to match changes in reader

### html

- Added "Quality" to the navigation list of links
- Fixed & Completed Data Quality
  - New dataQualityReport class
  - New resultFile class
- Fixed rangeElements -> rangeElement typo
- Fixed :class syntax errors

### iso19115-3

- Added distribution->liabilityStatement to the legal constraints
- Fixed & Completed Data Quality
  - Added Missing Fields
  - Fixed:
    - quality measure
    - evaluation method
    - descriptive result
    - quantitative result
  - Added:
    - conformance result
    - coverage result
    - quality result file
- Fixed rangeElements -> rangeElement typo
- Fixed date / datetime handling
- Removed duplicate code (class_organization)
- Added extended section with conditional to only write out if taxonomy exists
- Modified time period to print the date based on precision

### mdJson

- Fixed ordering in modules:
  - allocation
  - data quality
  - funding
  - legal constraint
  - metadata (data quality was at the bottom, should be after resourceInfo)
- Fixed & Completed Data Quality
  - New coverageResult module
  - Fixed & Completed Quality Measure
  - New qualityResultFile
  - New standaloneQualityReport (this was done incorrectly and none of the actual implementation was done)
- Fixed and added some checks for nil and empty

### sbJson

- Refactored citation to use otherDetails if present, otherwise only include authors and title
- External Contact Identifiers
  - Restricted to "person" contact type
  - Looks for external identifier with namespace "ORCID" and maps the associated `identifier` to `orcId`

### simple-html

- Added support for external contact identifiers
- Added support for Data Quality
  - New dataQualityReport section
  - New resultFile section
- Fixed rangeElement -> rangeElement typo
- Fixed :class syntax errors
- Fixed headers
- Fixed links
- Removed map
- Re-enabled scope extents (not sure why it was commented out here and not in the html writer)
- Bug fixes
  - fixed div that should be h5 (or other header tags)
  - fixed other details in citation
- Enhancements
  - Links
  - Headers
  - Map

## Fixed Unit Tests

- In the mdJson_hash_functions helpers for the tests
  - Fixed the alphabetized ordering for build_dataQuality
  - Fixed the ordering of the build_dataQuality output
- In the mdJson_hash_objects helpers for the tests
  - Fixed ordering of components
  - Fixed rangeElements -> rangeElement typo
  - Fixed dataQuality object

### reader - fgdc

- Fixed and added more assertions to lineage test
- Removed a couple process tests
- Fixed horizontal accuracy test (data quality)

### reader - iso19115-3

- Fixed id assertion
- Fixed date / datetime test data

### writer - mdJson

- Fixed order of data quality in metadata object
- New test runner for simple-html
- Initial test data for simple-html

jwaspin committed:
Update version

jwaspin committed:
Merge branch 'develop' into feature/sbjson-writer-external-contact-identifiers

jwaspin committed:
Add sbJson External Contact Identifiers

Reader and writer.
Reader looks for orcId and maps to external identifier.
Writer looks for external identifier with namespace ORCID and maps associated id to orcId.
Both only work for contact type of person, disabled for organizations.

jwaspin committed:
Add Data Quality Reports Array

jwaspin committed:
Fix internal metadata object

jwaspin committed:
Cleanup & Fix

jwaspin committed:
Fix Data Quality Reader - mdJson

jwaspin committed:
Cleanup

jwaspin committed:
Fix Data Quality

jwaspin committed:
Implement Data Quality Report

jwaspin committed:
Implement Coverage Result - mdJson writer

jwaspin committed:
Fix & Cleanup Data Quality

jwaspin committed:
Fix Spatial Representation

jwaspin committed:
Fix HTML citation Writer

jwaspin committed:
Cleanup

jwaspin committed:
Update Version

jwaspin committed:
Update Version

jwaspin committed:
Fix Internal Metadata Object - Data Quality Coverage Result

jwaspin committed:
Fix Refereence Typo

jwaspin committed:
Add Data Quality Link

jwaspin committed:
Handle Empty Citation

jwaspin committed:
Improve Data Quality

Add Reports Section

jwaspin committed:
Implement Data Quality Reports

jwaspin committed:
Add Null Checks

jwaspin committed:
Handle nil and empty

jwaspin committed:
Handle nil & empty

jwaspin committed:
New Generator newRangeElementDescription

J-Oliveros committed:
close out function

J-Oliveros committed:
fix reader to recognize rangeElement key from mdJson file (#317)

J-Oliveros committed:
fix key retrieval for Html_RangeElementDescription writer class (#318)

* fix key retrieval for Html_RangeElementDescription writer class

* fix br in rangeElement

J-Oliveros committed:
fix rangeElementDescription class for iso-3 (#321)

J-Oliveros committed:
add contact - external identifiers

jwaspin committed:
Merge branch 'feature/range-element-description' into releases/2-18-5

jwaspin committed:
Merge branch 'feature/sbjson-external-contact-identifier' into releases/2-18-5

jwaspin committed:
Fix Simple HTML and mdJson hash objects

rangeElements -> rangeElement

jwaspin committed:
Fix misplaced br tag

jwaspin committed:
Merge branch 'writers/html/data-quality-reports' into releases/2-18-5

J-Oliveros committed:
make changes based on feedback found in release v2.18.5 branch (#325)

* make changes based on feedback found in release v2.18.5 branch

* Fix evaluationMethod in mdJson

* remove extra argument

jwaspin committed:
Revert Changes

jwaspin committed:
Reorder and Add derivedElementReport

jwaspin committed:
Add Report Fields

type
standaloneQualityReportDetails

jwaspin committed:
Change dateTime to array

jwaspin committed:
Revert

jwaspin committed:
Add standaloneQualityReportDetails

jwaspin committed:
Add derivedElementReport

J-Oliveros committed:
add posacc w hori and vert tags. fix key retrieval from qualityMeasure (#327)

jwaspin committed:
New GitHub Test Action (#328)

This merge is not intended to have the tests passing yet, just adding the test action workflow file.

jwaspin committed:
Bugfix/unit tests (#329)

* Update Internal Object

* Fix Data Quality Order

* Fix Data Quality Order

* Fixing Legal Constraint Order

* Fix Allocation Order

* Fix Funding Order

* Fix Data Quality Order

* Update Data Quality Object

* And standaloneQualityReportDetails to mdJson Reader

* Fix Order

* Fix Standalone Quality Report

* Update Version for mdJson-schemas

* Add Quality Result File to Coverage Result

* Update Gemfile.lock

* Update nokogiri

* Rename standaloneQualityReport.rb

Need to fix naming case.

* Fix File Name

* Fix Typo

* Fix Internal Object

resultContent --> resultContentDescription and { } not [ ]

jwaspin committed:
Completed Feature: Coverage Result (#330)

* Add Quality Result File to Coverage Result

* add resultFile to coverageResult in mdJson reader (#331)

* add resultFile to coverageResult in mdJson reader

* remove pp lib

* revert files

* generate resourceFormat file and add to CR. fix metadata object in resultFile

* fix internal metadata obj in resultFile

* fix qualityResultFile class and reference

* revert files

* fix relative import

* remove writer changes

* Remove module_resourceFormat

* Injection/md json writer cr (#332)

* rename to qualityResultFile

* add format and QualityResultFile classes to coverageResult in writer

* revert files

* add qualityResultFile

* Fixed Names

fileName was repeated in place of fileDescription and fileType.

* Fix Result Content

resultContent -> resultContentDescription
Uses coverageDescription

* Change nil? to empty?

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Injection/html writer cr (#336)

* add qualityMeasure to mdJson reader

* fix qualityMeasure key

* add resultFile class to dataQualityReport html writer

* Fix Quality Measure

Identifier needs to be built not directly assigned.

* Improve Quality Measure

Name and description will only be populated if they exist.

---------

Co-authored-by: Jonathan Oliveros <joliveros@xentity.com>

* Injection/simple html writer cr (#335)

* add dataQualityReport and resultFile classes

* fix file format comment

* reference resultFile class

* change details to div

* change details to div in dataQualityReport

* change module to simple_html. Add nil or empty to citation within identifier.

* add simple_html test

* Add Coverage Result

New class_qualityResultFile for iso19115-3 writer.

* Fix Data Quality Report

Fixed the order.
Fixed quality measure.

* Fix Quality Measure

* Fix Quality Measure

* Fix Evaluation Method

* New Conformance Result ISO 19115-3 Writer

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>
Co-authored-by: Jonathan Oliveros <joliveros@xentity.com>

jwaspin committed:
Increment Minor Version

jwaspin committed:
Update Version

jwaspin committed:
Revert Changes

jwaspin committed:
Revert Changes

jwaspin committed:
Revert Changes

jwaspin committed:
Revert Changes

jwaspin committed:
Remove nil Check

Including the check for nil can actually cause errors. If there is an error then the problem is somewhere else.

jwaspin committed:
Remove nil Check

Should not check for nil, if it is nil then there's a problem with the reader.

jwaspin committed:
Revert Changes

jwaspin committed:
Update mdCodes Version

jwaspin committed:
Update Gemfile.lock

jwaspin committed:
Fix Data Quality

fgdc reader needed some refactoring.
Added/fixed checks for nil/empty.

jwaspin committed:
Fix name -> nameOfMeasure

jwaspin committed:
Increment Patch Version

jwaspin committed:
Fix FGDC Reader

quantitativeResult was being set to the qualityMeasure and using the qualityMeasure object. Fixed to use the correct object and location.

jwaspin committed:
Fix Data Quality (#342)

jwaspin committed:
Increment Patch Version

jwaspin committed:
Bugfixes for Simple HTML (#343)

* change divs to headers (#337)

* change divs to headers

* fix key within qualityMeasure. change to header tags

* Simple HTML Enhancements (#339)

* add geographicExtent to simple_html writer (#338)

* add geographicExtent to simple_html writer

* Fix Data Quality

Fixed checks for nil and empty.
Fixed tags.
Fixed structures.

* Fix Data Quality

* Remove bodyScript and other js

* Fix Typo referenceDocument -> referenceDocuments

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* add bookmark span tags to headers (#340)

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Increment Patch Version

* Increment Minor Version

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

jwaspin committed:
Increment Patch Version

jwaspin committed:
Fix Data Quality

jwaspin committed:
Increment Patch Version

jwaspin committed:
Update mdJson-schemas version

mdJson-schemas -> 2.8.3

jwaspin committed:
Increment Patch Version

jwaspin committed:
Update adiwg-mdtranslator gem dependencies

jwaspin committed:
Update adiwg-mdcodes gem dependency to version 2.9.1

jwaspin committed:
Disable test_mdJson_reader_schema_version_future_minor method

jwaspin committed:
Update mdtranslator version to 2.19.0-beta.6

jwaspin committed:
Update adiwg-mdcodes gem dependency to version 2.9.1

jwaspin committed:
Fix data quality report type

Remove DQ_ prefix.

jwaspin committed:
Update mdtranslator version to 2.19.0-beta.7

jwaspin committed:
Update adiwg-mdjson_schemas and adiwg-mdcodes gem dependencies

jwaspin committed:
Update adiwg-mdjson_schemas and adiwg-mdcodes gem dependencies

hmaier-fws commented:
@dwalt have you tested this version to verify that this PR closes the identified issues? What about the items identified under "*Might close the following*"? If only the primary issues have been addressed, I can merge in the PR and create a new issue to verify that the remaining items have been resolved.

dwalt commented:
Needs to be tested, pending

jwaspin committed:
Release Version 2.20.0 (#346)

* change divs to headers (#337)

* change divs to headers

* fix key within qualityMeasure. change to header tags

* Simple HTML Enhancements (#339)

* add geographicExtent to simple_html writer (#338)

* add geographicExtent to simple_html writer

* Fix Data Quality

Fixed checks for nil and empty.
Fixed tags.
Fixed structures.

* Fix Data Quality

* Remove bodyScript and other js

* Fix Typo referenceDocument -> referenceDocuments

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* add bookmark span tags to headers (#340)

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Increment Patch Version

* Increment Minor Version

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

* Feature/simple html (#345)

* add geographicExtent to simple_html writer (#338)

* add geographicExtent to simple_html writer

* Fix Data Quality

Fixed checks for nil and empty.
Fixed tags.
Fixed structures.

* Fix Data Quality

* Remove bodyScript and other js

* Fix Typo referenceDocument -> referenceDocuments

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* add bookmark span tags to headers (#340)

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Increment Patch Version

* Increment Minor Version

* Bugfixes for Simple HTML (#343)

* change divs to headers (#337)

* change divs to headers

* fix key within qualityMeasure. change to header tags

* Simple HTML Enhancements (#339)

* add geographicExtent to simple_html writer (#338)

* add geographicExtent to simple_html writer

* Fix Data Quality

Fixed checks for nil and empty.
Fixed tags.
Fixed structures.

* Fix Data Quality

* Remove bodyScript and other js

* Fix Typo referenceDocument -> referenceDocuments

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* add bookmark span tags to headers (#340)

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Increment Patch Version

* Increment Minor Version

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

* Increment Patch Version

* Fix Data Quality

* Increment Patch Version

* simple_html output formatting (#344)

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>

* Feature/fgdc reader lineage process (#347)

* add references to process

* redirect citation to processSteps

* revert to lineageCitation

* Fix newLineage Order

dataSources after processSteps

* Moving Methodology to Process Steps

* Restore hResourceInfo Variable

* Commented Keyword - Method Required Warning

* Upgrade adiwg-mdjson_schemas

adiwg-mdjson_schemas -> 2.9.0

* Update Gemfile.lock

* commented out mdJson reader minor version test

* Refactor lineage module and process module. Updated tests.

* Add comment about test being disabled.

* Add comment to explain why code is disabled

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>

* Update 19115-3 test data (#334)

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* upate attributeGroup

* update 19115_3_citation.xml

* update 19115_3_constraintCommon.xml

* update 19115_3_constraintCommon.xml and 19115_3_citation.xml

* update 19115_3_dataIdentification.xml

* update 19115_3_leProcessStep.xml

* update 19115_3_maintenance.xml

* update 19115_3_mdBand.xml

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update 19115-3 test data

* Update remoteSchema URL in class_mdMetadata.rb

* Add fix for 19115-3 writer in class_rangeElementDescription and class_sampleDimension

* fix unit definition identifiers in test cases

* fix attributeGroup test data

* Refactor algorithm element to LE_Processing in 19115_3_leProcessStep.xml and 19115_3_processing.xml

* Refactor algorithm element to LE_Processing in 19115_3_processing.xml

* Remove underscore from id's in iso19115-3 writer tests

* Update organization external identifier namespace from ORCID to ROR

Remove duplicate external contact identifier in iso19115-3 writer

* Update to ROR on feature xml and proper capitalization of multiPoint

* Removed partyIdentifier duplicate in expected organization output

* Remove duplicate partyIdentifier in expected constraintCommon xml

* Fix Spacing

* Fix spacing and remove duplicate partyIdentifier in expected constraintCommon xml

---------

Co-authored-by: Johnathan Aspinwall <jwaspin@gmail.com>
Co-authored-by: Jonathan Oliveros <joliveros@xentity.com>

* Fix version comparison bug in mdJson_reader.rb (#356)

* Fix version comparison bug in mdJson_reader.rb

* Fix version comparison logic

* Update adiwg-mdtranslator version to 2.20.0-beta.0

* Update adiwg-mdtranslator version to 2.19.0-beta.8

---------

Co-authored-by: J-Oliveros <97468338+J-Oliveros@users.noreply.github.com>
Co-authored-by: Chris MacDermaid <64213093+chris-macdermaid@users.noreply.github.com>
Co-authored-by: Jonathan Oliveros <joliveros@xentity.com>

jwaspin committed:
Update adiwg-mdjson_schemas and adiwg-mdcodes gem dependencies

jwaspin committed:
Update adiwg-mdcodes gem dependency to version 2.9.3.pre.beta.3

jwaspin committed:
Update adiwg-mdjson_schemas gem dependency to version 2.9.2.pre.beta.19

jwaspin committed:
Update mdtranslator version to 2.19.0-beta.9

jwaspin committed:
Update adiwg-mdtranslator version to 2.19.0.pre.beta.9

jwaspin committed:
Fix data quality report type

jwaspin committed:
Update adiwg-mdtranslator version to 2.19.0.pre.beta.10

jwaspin committed:
Fix evaluationMethodType in class_dataQualityReport.rb

jwaspin committed:
Update adiwg-mdjson_schemas and adiwg-mdcodes gem dependencies to latest versions

jwaspin committed:
Update adiwg-mdtranslator version to 2.19.0.pre.beta.11

jwaspin committed:
Update adiwg-mdtranslator and adiwg-mdjson_schemas gem dependencies to latest versions

jwaspin committed:
Fix evaluationMethodType in class_dataQualityReport.rb

jwaspin committed:
Fix data quality report type

jwaspin committed:
Update adiwg-mdtranslator version to 2.19.0.pre.beta.13

jwaspin committed:
Update adiwg-mdtranslator and adiwg-mdjson_schemas gem dependencies to latest versions

jwaspin committed:
Update adiwg-mdtranslator and adiwg-mdjson_schemas gem dependencies to latest versions

jwaspin committed:
Update adiwg-mdtranslator and adiwg-mdjson_schemas gem dependencies to latest versions

jwaspin committed:
Fix data quality report type

jwaspin committed:
Fix data quality report type

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.15

jwaspin committed:
chore: Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.15

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.16

jwaspin committed:
Update module files to include 'name' attribute

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.17

jwaspin committed:
Fix data quality report summary display

jwaspin committed:
Refactor data quality report type display

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.18

jwaspin committed:
Refactor mdJson_onlineResource module to include 'protocol' attribute

jwaspin committed:
Refactor mdJson_resourceInfo module to include 'credit' attribute

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.18

jwaspin committed:
Refactor data quality report type display

jwaspin committed:
chore: Update versions

mdJson-schemas to v2.9.3-beta.3
mdCodes to v2.9.4-beta.4

jwaspin committed:
chore: Increment beta version

jwaspin committed:
Revert Data Quality Report Types

jwaspin committed:
Refactor sbJson_citation.rb

1. If the user provided Other Cit Details, then use that
2. If Other Cit Details is null, then provide a basic citation of: author1, author2, ..., Citation Title

jwaspin committed:
Refactor data quality report type display and add system identifier support

jwaspin committed:
Refactor data quality report logic handling

jwaspin committed:
Refactor data quality reports

jwaspin committed:
Refactor data quality reports

jwaspin committed:
chore: Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.21

jwaspin committed:
Refactor data quality reports

jwaspin committed:
chore: Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.22

jwaspin committed:
Refactor XML date format in test data files

jwaspin committed:
Refactor data identification to handle distribution liability statements

jwaspin committed:
chore: Update adiwg-mdtranslator gem dependency to version 2.19.0.pre.beta.25

jwaspin committed:
Update adiwg-mdtranslator gem dependency to version 2.19.0

jwaspin committed:
Add 'Attribute Accuracy Explanation' text in the evaluation method name field for data quality.

jwaspin committed:
Refactor data quality reports to handle multiple results and different report types

jwaspin committed:
Fix altmax to use maximumAltitude

jwaspin committed:
Refactor Positional Accuracy Reports

jwaspin committed:
fix: Unit test for maximum altitude

jwaspin committed:
Merge branch 'releases/2-19-0' of github.com:adiwg/mdTranslator into releases/2-19-0

jwaspin committed:
Refactor unit test for positional accuracy

jwaspin committed:
feat: Add metadata extension section for taxonomy info.

jwaspin committed:
feat: Set conditional for metadata extension taxonomy.

jwaspin committed:
feat: Remove metadata extension for taxonomy information

jwaspin committed:
Fix: Replaced div with h5 as needed

jwaspin committed:
fix: Citation Other Details

jwaspin committed:
Refactor class_timePeriod.rb for improved date handling

jwaspin committed:
chore: Add note to move object out of function

jwaspin committed:
chore: Refactor unit tests to match modifications to dateTime in timePeriod

jwaspin committed:
chore: Update mdtranslator version to 2.19.0-beta.31

jwaspin committed:
chore: Update adiwg-mdtranslator version to 2.19.0.pre.beta.31

hmaier-fws reviewed (approved):
Has been published to test site for a long time with no complaints.
